validation/sequence_provider.rst

```diff
@@ -282,10 +282,22 @@ method, which should return an array of groups to use::
                 $groups[] = 'Premium';
             }
 
+            // Caution: the validation behavior depends on how this method
+            // returns its result (a simple array or a nested array).
+
+            // if the User group causes any violations, the Premium group will
+            // not be validated (even if $this->isPremium() is true)
             return $groups;
+
+            // both groups are validated and you'll get violations from any of them
```

Comment:
Technically, this is correct. But I guess that readers might be confused about why they should use a group sequence for this example at all then since now validation constraints from all groups would be evaluated. I think we need a better example to show how to use a list of groups in one group sequence step.

Maybe something like `[['User', 'Premium'], 'Api']`, so that both the "User" and the "Premium" group will be evaluated in the first step (with "Premium" only if the user is a premium user) and the costly "Api" check will only be performed if none of the previous groups triggered a violation.

Reply:
Your example is much better. I've reworded everything. Thanks!